ci/steps/notify.py

```diff
@@ -0,0 +1,136 @@
+import ccc.github
+import ci.util
+import distutils.util
+from email import encoders
+from email.mime.base import MIMEBase
+import glci.notify
+import glci.util
+import json
+import mailutil
+import os
+from string import Template
+import typing
+import urllib
+
+
+def _email_cfg(cicd_cfg: glci.model.CicdCfg):
+    ctx = ci.util.ctx()
+    cfg_factory = ctx.cfg_factory()
+    return cfg_factory.email(cfg_name=cicd_cfg.notify.email_cfg_name)
+
+
+def _smtp_client(email_cfg):
+    return glci.notify.smtp_client(
+        host=email_cfg.smtp_host(),
+        user=email_cfg.credentials().username(),
+        passwd=email_cfg.credentials().passwd(),
+)
+
+
+def send_notification(
+    cicd_cfg_name: str,
+    giturl: str,
+    status_dict_str: str,
+    additional_recipients: typing.Sequence[str] = [],
+):
+    namespace = '$(params.namespace)'
+    pipeline_name = '$(params.pipeline_name)'
+    pipeline_run = '$(params.pipeline_run_name)'
+
+    repo_dir = '$(params.repo_dir)'
+
+    if distutils.util.strtobool('$(params.disable_notification)'):
+        print('Notification is disabled, bot sending email')
+        return
```

Comment:
These should be passed using the existing Framework

Reply:
not sure about this comment: it uses glci.notify....